sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -343,3 +343,6 @@ def get_keyed_state(self, key):
     if not self.partial_keyed_state.get(key):
       self.partial_keyed_state[key] = self.existing_keyed_state[key].clone()
     return self.partial_keyed_state[key]
+
+  def clear_partial_states(self):
```

Comment:
<!--thread_id:cc_139172690_t; commit:af94b7b7a8c0ac0ada222884101ec10127f5b234; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> I think it would be better to have a `reset()` method on `_ExecutionContext` that can be called to clear the step context and remove `clear_partial_states()`.  We can then call this in `attempt_call()`.

<!--section:body-->
Done.